File: Mockingbird-Objective-C/Mockingbird-Objective-C/ViewController.m
//
//  ViewController.m
//  Mockingbird-Objective-C
//
//  Created by xiangwenwen on 15/6/17.
//  Copyright (c) 2015年 xiangwenwen. All rights reserved.
//

#import "ViewController.h"
#import "Mockingbird_Objective_C-Swift.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *showScanCodeInfo;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.showScanCodeInfo.numberOfLines = 0;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)scanCode:(UIButton *)sender {
    UIStoryboard *stor = [UIStoryboard storyboardWithName:@"Mockingbird" bundle:[NSBundle mainBundle]];
    UINavigationController *navc =  [stor instantiateViewControllerWithIdentifier:@"MockingbirdNavigationID"];
    MockingbirdScanCodeManager *mocking = (MockingbirdScanCodeManager *)navc.topViewController;
    mocking.mockingbirdResult = ^(NSString * __nonnull value){
        self.showScanCodeInfo.text = value;
    };
    [self presentViewController:navc animated:YES completion:nil];
}

@end

File: MockingbirdDemo/ViewController.swift
//
//  ViewController.swift
//  Mockingbird
//
//  Created by xiangwenwen on 15/6/15.
//  Copyright (c) 2015年 xiangwenwen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var scancode: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "mockingbird:", name:MOKNotifiScanResult, object: nil)
        
    }
    
   func mockingbird(note:NSNotification)->Void{
        if let userInfo:Dictionary<NSObject,AnyObject> = note.userInfo{
            self.scancode.text = userInfo["value"] as? String
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func scancodeinfo(sender: UIButton) {
        let storyboard:UIStoryboard = UIStoryboard(name: "Mockingbird", bundle: NSBundle.mainBundle())
        let navC:UINavigationController = storyboard.instantiateViewControllerWithIdentifier("MockingbirdNavigationID") as! UINavigationController
        let mockingbird:MockingbirdScanCodeManager = navC.topViewController as! MockingbirdScanCodeManager
        mockingbird.globalTitle = "扫描条码"
        mockingbird.globalColor = MOKTeal
        self.presentViewController(navC, animated: true, completion: nil)
    }
}

